### PR TITLE
[WebAssembly] Add intrinsics to wasm_simd128.h for all FP16 instructions

### PR DESCRIPTION
**Pull request by brendandahl**
Getting this to work required a few additional changes:
 - Add builtins for any instructions that can't be done with plain C currently.
 - Add support for the saturating version of fp_to_<s,i>_I16x8. Other vector sizes supported this already.
 - Support bitcast of f16x8 to v128. Needed to return a __f16x8 as v128_t.

### CODE REVIEW

File: llvm/lib/Target/WebAssembly/WebAssemblyInstrSIMD.td [hunk 1]
```diff
@@ -165,8 +165,9 @@ def F16x8 : Vec {
  let prefix = "f16x8";
 }
 
-// TODO: Include F16x8 here when half precision is better supported.
-defvar AllVecs = [I8x16, I16x8, I32x4, I64x2, F32x4, F64x2];
+// TODO: Remove StdVecs when the F16x8 works every where StdVecs is used.
```
Comment (brendandahl) [inline llvm/lib/Target/WebAssembly/WebAssemblyInstrSIMD.td:168]: It's not obvious from this patch, but now `AllVecs` is only used in one place for bitcast (which means it now works for f16x8 vectors too).

Alternatively, I can leave `AllVecs` alone and just concat F16x8 down where bitcast is supported.

**Comment by sunfishcode** [comment]
Would it make sense to put these declarations behind `#ifdef __wasm_fp16__` so that they aren't declared if fp16 support isn't enabled?

**Comment by brendandahl** [comment]
> Would it make sense to put these declarations behind `#ifdef __wasm_fp16__` so that they aren't declared if fp16 support isn't enabled?

I could do that, if that's preferred. I followed what the relaxed instructions did and use the target attribute `__target__("fp16")`.

**Comment by sunfishcode** [comment]
Oh, I missed that. In that case, that seems sufficient. Thanks!

**Comment by aheejin** [comment]
Looks good to me, but I'm not an expert here.. Maybe @tlively can take a look?

**Review by tlively — APPROVED** [review]
Looks great! Glad you found the cross-project-tests.

**Pull request merged**